atlas_densities/densities/inhibitory_neuron_densities_optimization.py

```diff
@@ -199,16 +199,20 @@ def _compute_region_cell_counts(
              ...                             ...
         The index is the sorted list of all region identifiers.
     """
-    id_counts = []
-    for id_ in tqdm(hierarchy_info.index):
-        mask = annotation == id_
-        id_counts.append(np.sum(density[mask]) * voxel_volume)
+    vtiv = voxel_data.ValueToIndexVoxels(annotation)
+    density_copy = vtiv.ravel(density.copy())
 
+    id_counts = []
+    for id_ in hierarchy_info.index:
+        indices = vtiv.value_to_1d_indices(id_)
+        if len(indices):
+            id_counts.append(np.sum(density_copy[indices]) * voxel_volume)
+        else:
+            id_counts.append(0)
```

Comment:
Is the check here needed?
If the indices are an empty array, then the sum on the empty slice will return 0, won't it?

Reply:
yup, good point, removed.